test/jdk/java/awt/Frame/DeiconifyClipTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2003, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * DeiconifyClipTest.java
+ *
+ * summary:
+ *
+ * What happens is that we call AwtWindow::UpdateInsets when
+ * processing WM_NCCALCSIZE delivered on programmatic deiconification.
+ * At this point IsIconic returns false (so UpdateInsets proceeds),
+ * but the rect sizes still seems to be those weird of the iconic
+ * state.  Based on them we compute insets with top = left = 0 (and
+ * bottom and right that are completely bogus) and pass them to
+ * PaintUpdateRgn which results in incorrect clip origin.  Immediately
+ * after that we do UpdateInsets again during WM_SIZE processing and
+ * get real values.
+ */
+
+import javax.swing.BoxLayout;
+import javax.swing.JFrame;
```

Comment:
Since it is an AWT test should we be using Frame instead of JFrame ?

Reply:
I guess this is not a big deal in this case.